lib/pbkdf2.ex

```diff
@@ -153,11 +153,20 @@ defmodule Pbkdf2 do
   """
   @impl true
   def verify_pass(password, stored_hash) do
-    [alg, rounds, salt, hash] = String.split(stored_hash, "$", trim: true)
-    digest = if alg =~ "sha512", do: :sha512, else: :sha256
+    [prefix, rounds, salt, hash] = String.split(stored_hash, "$", trim: true)
+    digest = hash_prefix_to_digest(prefix)
+
     Base.verify_pass(password, hash, salt, digest, rounds, output(stored_hash))
   end
 
+  defp hash_prefix_to_digest("pbkdf2" <> _ = prefix) do
+    prefix
+    |> String.slice(7..-1)
+    |> String.to_existing_atom()
+  end
+
+  defp hash_prefix_to_digest(digest), do: String.to_existing_atom(digest)
```

Comment:
Is it safe here to assume that the digest atom exists?

Reply:
I believe it is safe.
Calling any function from `:crypto` inside a module should load that module.
E.g.
```
iex(1)> String.to_existing_atom("sha224")
** (ArgumentError) argument error
    :erlang.binary_to_existing_atom("sha224", :utf8)
iex(1)> :crypto.rand_uniform(1, 2)
1
iex(2)> String.to_existing_atom("sha224")
:sha224
```
This module has calls to crypto so it is safe to assume that during runtime `:crypto` is loaded with its atoms.